src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -369,34 +367,31 @@ private synchronized SecretKey getKeyOrCreate(final String keyVersion) throws Ge
      * @throws IOException
      */
     private synchronized SecretKey getKey(final String keyVersion) throws GeneralSecurityException, IOException {
-        if (VERSION_USER_DEFINED.equals(keyVersion)) {
+        switch (keyVersion) {
+        case VERSION_USER_DEFINED : 
             return getSecretKey(AuthenticationSettings.INSTANCE.getSecretKeyData());
-        }
-        
-        if (VERSION_ANDROID_KEY_STORE.equals(keyVersion)) {
+        case VERSION_ANDROID_KEY_STORE :
             if (Build.VERSION.SDK_INT >= 18) {
```

Comment:
ah ok. Sorry, I missed it. Yeah, we talked about it, it's not possible for user to downgrade their device, so this is not likely to happen. Will update.